store/stats.ts

```diff
@@ -0,0 +1,105 @@
+import { getStats } from '../util/namex-api'
+
+export enum StatsColumns {
+  RequestDetails = 'Request Details',
+  Status = 'Status',
+  LastUpdate = 'Last Update',
+  BackToNRO = 'Updated to NRO'
+}
+
+/**
+ * Fetches paged statistics based on the provided parameters.
+ * @param {string} timespan - The timespan for the statistics.
+ * @param {boolean} myStats - Indicates whether to fetch the user's statistics.
+ * @return {Promise<any>} A Promise that resolves to the fetched statistics data.
+ * @throws {Error} Throws an error if the network response is not successful.
+ */
+export async function getPagedStats (timespan: string | null, myStats: boolean) {
+  let params = ''
+  if (timespan !== '' && timespan !== null) {
+    params = '?timespan=' + timespan
+  } else {
+    params = '?timespan=1'
+  }
+  // TODO the numbers should not be fixed
+  params += '&currentpage=1&perpage=1000'
+  if (myStats) {
+    params += '&myStats=True'
+  }
+
+  try {
+    const response = await getStats(params)
+
+    if (response.ok) {
+      const jsonData = await response.json() // Parse response JSON
+
+      if (jsonData.numRecords >= 0) {
+        const requestsData = jsonData.nameRequests
+
+        const processedData = requestsData.map((statsData:any) => {
+          statsData.names = sortNames(statsData.names)
+          if (statsData.comments.length === 1 && statsData.comments[0].timestamp) {
+            statsData.comments[0].timestamp = convertDate(statsData.comments[0].timestamp)
+          }
+          statsData.lastUpdate = convertDate(statsData.lastUpdate)
+          statsData.names.forEach((name:any) => {
+            if (name.decision_text) {
+              name.decision_text = name.decision_text.split(/\n\n/)
+            }
+          })
+          return statsData
+        })
+
+        processedData.numRecords = jsonData.numRecords
```

Comment:
Can the length of `processedData` just be used by the caller? This is adding a field to an `Array` object.